File: StreamDeck_Tests/Teleop_CPP/data_publisher.cxx
#include <iostream>

#include <dds/pub/ddspub.hpp>
#include <rti/util/util.hpp>     
#include <rti/config/Logger.hpp> 

#include "data.hpp"

void run_publisher_application(){
    
    dds::domain::DomainParticipant participant(0);

    dds::topic::Topic< ::statistic_data> topic(participant, "Example statistic_data");

    dds::pub::Publisher publisher(participant);

    dds::pub::DataWriter< ::statistic_data> writer(publisher, topic);

    ::statistic_data data;
    
   while(true){
        
        data.auto_flag(static_cast< int16_t>(1));
        std::cout << "Writing ::statistic_data" << std::endl;

        writer.write(data);

        rti::util::sleep(dds::core::Duration(1));
    }
}

int main(int argc, char *argv[])
{

    run_publisher_application();
    dds::domain::DomainParticipant::finalize_participant_factory();

    return EXIT_SUCCESS;
}

File: INTEGRATION_GO/Streamdeck_side/data_publisher.py

# (c) Copyright, Real-Time Innovations, 2022.  All rights reserved.
# RTI grants Licensee a license to use, modify, compile, and create derivative
# works of the software solely for use with RTI Connext DDS. Licensee may
# redistribute copies of the software provided that all such copies are subject
# to this license. The software is provided "as is", with no warranty of any
# type, including any warranty for fitness for any purpose. RTI is under no
# obligation to maintain or support the software. RTI shall not be liable for
# any incidental or consequential damages arising out of the use or inability
# to use the software.

import time
import sys
import rti.connextdds as dds
from data import IMU_data

class IMU_dataPublisher:

    @staticmethod
    def run_publisher(domain_id: int, sample_count: int):

        # A DomainParticipant allows an application to begin communicating in
        # a DDS domain. Typically there is one DomainParticipant per application.
        # DomainParticipant QoS is configured in USER_QOS_PROFILES.xml
        participant = dds.DomainParticipant(domain_id)

        # A Topic has a name and a datatype.
        topic = dds.Topic(participant, "Example IMU_data", IMU_data)

        # This DataWriter will write data on Topic "Example IMU_data"
        # DataWriter QoS is configured in USER_QOS_PROFILES.xml
        writer = dds.DataWriter(participant.implicit_publisher, topic)
        sample = IMU_data()        

        for count in range(sample_count):
            # Catch control-C interrupt
            try:
                # Modify the data to be sent here
                
                print(f"Writing IMU_data, count {count}")
                writer.write(sample)
                time.sleep(1)
            except KeyboardInterrupt:
                break

        print("preparing to shut down...")


if __name__ == "__main__":
    IMU_dataPublisher.run_publisher(
            domain_id=0,
            sample_count=sys.maxsize)
